pkgs/by-name/ki/kimai/package.nix

```diff
@@ -0,0 +1,54 @@
+{pkgs,
+system ? builtins.currentSystem,
+config,
+consoleCmd ? "kimai-console",
+databaseUrl ? "",
+mailerFrom ? "",
+mailerUrl ? "",
+appSecret ? "",
+corsAllowOrigin ? "",
+...
+}:
+
+let
+  phpPackage = import ./composer2nix/default.nix {
+    inherit pkgs system;
+  };
+in
+phpPackage.override {
+  postInstall = ''
+    # ensure the dirs exist, just in case,
+    # move any initialized data to "sample" to make space for symlinks.
+    # Finally, symlink to a directory outside of the nix store so that kimai
+    # may write to them without issue.
+    # These symlinks are created in the nix store but their targets need to be created later on,
+    # when the service starts.
+
+    mkdir -p ./var/cache
+    mkdir -p ./var/log
+    mkdir -p ./var/data
+    mv ./var/cache ./var/cache.sample
+    mv ./var/log ./var/log.sample
+    mv ./var/data ./var/data.sample
+    ln -s /var/lib/kimai/var/cache ./var/cache
+    ln -s /var/lib/kimai/var/log ./var/log
+    ln -s /var/lib/kimai/var/data ./var/data
+    ln -s /var/lib/kimai/var/plugins ./var/plugins
```

Comment:
Doesn't this result in an impure package? It's altering stuff outside of its location in the nix store.

Instead this should be handled by a systemd service in a module that does this (maybe by setting the current working dir while running it or doing a bind mount of the source folders to a runtime directory).

Reply:
I agree these paths should be configurable